packages/compiler-cli/test/compliance/test_cases/r3_view_compiler_template/nested_template_context_many_bindings_template.js

```diff
@@ -4,8 +4,8 @@ function MyComponent_div_0_Template(rf, ctx) {
     $r3$.ɵɵelementStart(0, "div", 1);
     $r3$.ɵɵlistener("click", function MyComponent_div_0_Template_div_click_0_listener() {
       $r3$.ɵɵrestoreView($s$);
-      const $d$ = ctx.$implicit;
-      const $i$ = ctx.index;
+      const $d$ = $r3$.ɵɵgetContext($s$).$implicit;
+      const $i$ = $r3$.ɵɵgetContext($s$).index;
```

Comment:
```suggestion
      const ctx = $r3$.ɵɵgetContext($s$);
      const $d$ = ctx.$implicit;
      const $i$ = ctx.index;
```

We should store it in local var, rather than creating the call instruction multiple times.

Reply:
Come to think of it can we just change `ɵɵrestoreView` to return `ctx`
```
  const ctx = $r3$.ɵɵrestoreView($s$);
  const $d$ = ctx.$implicit;
  const $i$ = ctx.index;
```